src/PowerShellEditorServices.Protocol/LanguageServer/EditorCommands.cs

```diff
@@ -132,8 +132,15 @@ public static readonly
     public class SaveFileRequest
     {
         public static readonly
-            RequestType<string, EditorCommandResponse, object, object> Type =
-            RequestType<string, EditorCommandResponse, object, object>.Create("editor/saveFile");
+            RequestType<SaveFileDetails, EditorCommandResponse, object, object> Type =
+            RequestType<SaveFileDetails, EditorCommandResponse, object, object>.Create("editor/saveFile");
+    }
```

Comment:
Would this be a breaking change?  I doubt anyone is using this request directly, but it's worth noting `IMessageSender` is part of the public API as it's available from `$psEditor.Components`

Reply:
Good point. This will be a part of a 1.7.0 release so we'll just have to be explicit and say that there's a breaking change.

Reply:
unless there's a way to be backcompat friendly?

Reply:
Good point -- we really should start thinking of the API itself as the contract, not the VSCode behaviour.

I'll change it back for the next 1.x release and then we can change it back again for 2.0. Does that sound reasonable?

(We don't really need to change it for 2.0 -- was just going for a consistent minimal API)

Reply:
breaking changes this small aren't bad. We just need to make sure we let people know in the release notes.

Reply:
Yeah ok -- I think that's fair. Especially after all the people asking for `SaveAs()`.

Reply:
Yeah definitely not a big deal.  I highly doubt there is a single person using that request directly with `IMessageSender` right now.  Just something we should really keep an eye on as PSES is just starting to see some adoption by other folks.

I could see `SaveAs` being a separate request if you wanted to be sure we don't mess anyone up, but I'm not sure how much refactoring that'd be.  Probably not worth it.

Reply:
Yeah, I was thinking about setting it back to use a different request type, but now @tylerl0706 has convinced me that it's more effort than it's worth. But definitely agree that we need to keep the API stable.

I remember looking at the `Save()` API and thinking whether `SaveAs()` should be implemented independently and deciding that it makes sense (in a world without downstream developers) as a single field.

In general though, one thing I think might be worth doing is changing any API calls that we might want to extend which use a primitive (e.g. `string`) body to using an object body with a field, just so that we can conceivably add a new field and break the API "less"...

Reply:
That last paragraph is awesome. Reminds me of graphQL in a lot of ways

Reply:
*shudder*